sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -30,42 +30,40 @@
 #define kIllegalAddr1 0x4041FFF0u
 #define kIllegalAddr2 0x40003618u
 #define kSkipComp 0x12345678u
-#define kCpuDumpSize 9
+#define kCpuDumpSize 8
 
 /**
  * Cpu dump struct index
  */
 enum {
   kCpuDumpIdxCurrentExceptionAddr = 0,
-  kCpuDumpIdxCurrentLastDataAddr = 1,
-  kCpuDumpIdxCurrentNextPc = 2,
-  kCpuDumpIdxCurrentPc = 3,
-  kCpuDumpIdxPreviousExceptionAddr = 4,
-  kCpuDumpIdxPreviousLastDataAddr = 5,
-  kCpuDumpIdxPreviousNextPc = 6,
-  kCpuDumpIdxPreviousPc = 7,
-  kCpuDumpIdxPreviousValid = 8,
+  kCpuDumpIdxCurrentExceptionPc = 1,
+  kCpuDumpIdxCurrentLastDataAddr = 2,
+  kCpuDumpIdxCurrentNextPc = 3,
+  kCpuDumpIdxCurrentPc = 4,
+  kCpuDumpIdxPreviousExceptionAddr = 5,
+  kCpuDumpIdxPreviousExceptionPc = 6,
+  kCpuDumpIdxPreviousValid = 7,
 };
 
 /**
  * Reserve expected cpu dump area in flash
  */
 __attribute__((section(".non_volatile_scratch")))
 const volatile dif_rstmgr_cpu_info_dump_segment_t exp_dump[kCpuDumpSize] = {
-    UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX,
+    UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX,
     UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX};
 
 /**
  *  Dump structure:
- *    0: current.exception_addr
- *    1: current.last_data_addr
- *   *2: current.next_pc
- *   *3: current.pc
- *    4: previous.exception_addr
- *    5: previous.last_data_addr
- *   *6: previous.next_pc
- *    7: previous.pc
- *    8: previous_valid
+ *    0: current.exception_pc
```

Comment:
Comparing this description with the enum in the line 38, this line seems to be inverted with the next.

Reply:
yes good catch.